src/odemis/driver/orsay.py

```diff
@@ -3039,12 +3037,34 @@ def __init__(self, name, role, parent, **kwargs):
         self.probeCurrent = model.VAEnumerated(current, unit="A",
                                                choices=self._probe_current_presets.keys() | {None},
                                                setter=self._setProbeCurrent)
+        self.probeCurrent.subscribe(self._onProbeCurrent, init=True)
 
         # TODO: it seems that the standard client store the latest preset loaded in
         # datamodel.IonColumnMCS.ConditionName. Setting .Target = PresetName
         # seems to not bother anything else. So maybe we should do that, and rely
         # on it to guess the preset in use (and so probe current).
 
+        # The .shift represents fib_beam.shift but with the value being independent
+        # from the probe current. The actual fib_beam.shift has to
+        # be adjusted for each probe current, so that the center of the FoV
+        # corresponds to the same point. Also, every time the probe current is
+        # changed, it's reset. We don't want to pass this trouble to the higher
+        # level code. So the "base" shift for each probe current is stored (and
+        # shared via MD_SCAN_OFFSET_CALIB), and it's hidden here, so that a
+        # shift of N µm always stays, even if the current changes.
```

Comment:
what do you mean with "a shift of N µm always stays"?

Reply:
I mean that if the user change .shift to a value (x, y) (that I called `N`), and then changes other settings on the FIB, like the current, the shift value will stay (x, y). Maybe that sounds obvious, but so far, the behaviour was that the shift would be reset after changing the current... to some arbitrary values which correspond to "no shift".

Reply:
here is my attempt to make it clearer:
```
        # The .shift represents fib_beam.shift but with the value being
        # independent from the probe current. The actual fib_beam.shift has to
        # be adjusted for each probe current, so that the center of the FoV
        # corresponds to the same point. Changing the probe current in the Orsay
        # API, automatically reset the beam to the preset values (which should
        # correspond to a shift of 0, 0). We hide this confusing behaviour by
        # only storing the shift which is extra to the preset shift (aka "base"
        # shift). Every time the probe current is changed, the fib_beam.shift is
        # adjusted to keep the extra shift correct. That means the user can set
        # a shift of (x, y) µm, and it will always stay, even if the current
        # changes. The "base" shifts for each probe current are read from the
        # presets and stored in MD_SCAN_OFFSET_CALIB. Note: if the Orsay beam
        # shift is modified on the Orsay side, this shift is not updated (while
        # fib_beam.shift is updated). Making it automatically update would be a
        # lot more work/complexity to make sure that we 100% are certain it's
        # due to a manual change and not just reset after changing probe
        # current. Also, the probeCurrent also doesn't update automatically, so
        # it's the same behaviour.
 ```

Reply:
Thanks I understand it now. Maybe you could make it even more explicit:
```
        That means the user can set a shift of (x, y) µm,
        and it will always stay (x, y) µm, even if the current changes.
```